modules/photons_control/clean.py

```diff
@@ -0,0 +1,188 @@
+from photons_control.script import FromGenerator
+
+from photons_app.tasks import task_register as task
+from photons_app.errors import PhotonsAppError
+
+from photons_messages import LightMessages, DeviceMessages
+
+
+def SetCleanConfig(indication=None, duration=None, **kwargs):
+    """
+    Returns a valid message that will either get or set the default clean cycle
+    configuration for the device.
+
+    For example:
+
+    .. code-block:: python
+
+        await target.send(CleanConfig(indication=True), ["d073d5000001", "d073d5000001"])
+    """
+
+    async def gen(reference, sender, **kwargs):
+        has_hev = DeviceMessages.GetVersion()
+        async for pkt in sender(has_hev, reference, **kwargs):
+            if pkt | DeviceMessages.StateVersion:
+                if pkt.product == 90 or pkt.product == 99:
+                    yield LightMessages.SetHevCycleConfiguration(
+                        indication=True if indication else False,
+                        duration_s=duration if duration else 7200,
+                        target=pkt.serial,
+                        res_required=False,
+                    )
+
+    return FromGenerator(gen)
+
+
+@task
+class set_clean_default_config(task.Task):
+    """
+    Set the default clean cycle configuration for a device
+
+    ``target:clean_config match:cap=hev -- '{"indication": true, "duration": 7200}'``
+
+    It takes an "indication" field that is a boolen which indicates whether to
+    run a short flashing indication at the end of the HEV cycle. It also takes
+    a ``duration`` field that is the seconds of the duration.
+    """
+
+    target = task.requires_target()
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        options = self.photons_app.extra_as_json
+
+        if "indication" not in options and "duration" not in options:
```

Comment:
this is where the norms thing comes in handy https://delfick-project.readthedocs.io/en/latest/api/norms/index.html

(I really want to[ fix the API ](https://github.com/delfick/delfick_project/blob/ideas/rewrite-norm-api/IDEAS.rst)though but that is probably many more months down the track)

But for now you can do this

```
from delfick_project.norms import sb, Meta


options = self.photons_app.extra_as_json
options = sb.set_options(indication=sb.required(sb.boolean()), duration=sb.required(sb.float_spec())).normalise(Meta.empty(), options)
```

And then you are guaranteed that options only has indication and duration and they are the correct types

Reply:
D'oh, of couse.